test/metabase/lib/remove_replace_test.cljc

```diff
@@ -1423,3 +1426,46 @@
             (lib/replace-clause join-query 0
                                 (first (lib/aggregations join-query 0))
                                 (lib/min (lib/get-month (by-name (lib/orderable-columns join-query 0) "CREATED_AT"))))))))
+
+(deftest ^:parallel coerced-filter-test
+  (let [effective-type (types/effective-type-for-coercion :Coercion/UNIXSeconds->DateTime)
+        mp (merged-mock/merged-mock-metadata-provider
+            meta/metadata-provider
+            {:fields [{:id                (meta/id :people :id)
+                       :coercion-strategy :Coercion/UNIXSeconds->DateTime
+                       :effective-type    effective-type}]})
+        old-clause [:=
```

Comment:
You should build clauses and queries with `lib` functions
```
(-> (lib/query meta/metadata-provider (meta/table-metadata :people)
  (lib/filter (lib/= (meta/field-metadata :people :city) "Abilene")))
```


and use `lib/filters` to get a reference to `old-clause`

If you need to drop `effective-type` from a clause use an `update-in [:filters 1 ...]` to drop it from the query.

Reply:
Thanks, I'll use lib functions to build queries in tests going forward.